hw_01.js

```diff
@@ -0,0 +1,92 @@
+// 1. Задача на повернення ініціалів для кожного імені з масиву, посортованих в алфавітному порядку: 
+
+// const userNames = ["Петрик Ольга Іванівна", "Гнатюк Петро Антонович", "Рудко Андрій Опанасович"];
+// let initials = [];
+// for (const userName of userNames) {
+//     const partNames = userName.split(" ");
+//     let stringInitials = "";
+
+//     for (const part of partNames) {
+
+//         stringInitials = stringInitials + part[0] + ".";
+//     }
+//     initials.push(stringInitials);
+//     initials.sort();
+
+// }
+// console.log(initials);
+
+// З використанням методу 'map'.
+const userNames = ["Петрик Ольга Іванівна", "Гнатюк Петро Антонович", "Рудко Андрій Опанасович"];
+
+const initials = userNames.map(userName =>
+    userName.split(" ").map(part => part[0] + ".").join(""));
+initials.sort();
+console.log(initials);
+
+// 2. Задача на фільтрування масиву:
+// 2.1.1 Через умовну конструкцію:
+// const userNames = ['Петро', 'Емма', 'Юстин', 'Ілля', 'Марта', 'Яна', 'Василь', 'Антон', 'Олена'];
+// const filteredNames = [];
+
+// for (let i = 0; i < userNames.length; i++) {
+//     const userName = userNames[i];
+//     const firstLetter = userName[0];
+
+//     const vowelLetter = ['У', 'Е', 'І', 'А', 'О', 'Є', 'Я', 'Ю', 'И'];
+//     if (vowelLetter.includes(firstLetter)) {
+//         filteredNames.push(userName);
+//     }
+// }
+// console.log(filteredNames);
+
+// 2.1.2 За допомогою методу "forEach":
+// const userNames = ['Петро', 'Емма', 'Юстин', 'Ілля', 'Марта', 'Яна', 'Василь', 'Антон', 'Олена'];
+// const filteredNames = [];
+// userNames.forEach(userName => {
+//     const firstLetter = userName[0];
+//     const vowelLetter = ['У', 'Е', 'І', 'А', 'О', 'Є', 'Я', 'Ю', 'И'];
```

Comment:
```suggestion
// const filteredNames = [];
//     const vowelLetter = ['У', 'Е', 'І', 'А', 'О', 'Є', 'Я', 'Ю', 'И'];
// userNames.forEach(userName => {
//     const firstLetter = userName[0];
```
Зараз у вас на кожній ітерації, під капотом, створюється новий масив. Що звісно не оптимально

Його треба винести назовні з методу, тоді не буде лишнього використання пам'яті

Reply:
Винесла масив з методу.

Reply:
Винесла масив з методу.